### ISSUE TITLE
`TestTeleportClient_Login_local/OTP_device_login_with_hijack` flakiness

### ISSUE THREAD

**Issue by strideynet** [gravitational/teleport#16037]
#### Link(s) to logs
- https://console.cloud.google.com/cloud-build/builds/228ddfac-29a3-4570-a797-a96ff6c9679c?project=ci-account

#### Relevant snippet

```
exit status 1
time="2022-08-31T18:05:23Z" level=fatal msg="FAILED: unit tests failed\n\texit status 2"
time="2022-08-31T18:05:23Z" level=info msg="Removing data dir"
time="2022-08-31T18:05:23Z" level=info msg="Waiting for etcd process to exit..."
time="2022-08-31T18:05:23Z" level=info msg="Killing etcd process"
time="2022-08-31T18:05:23Z" level=info msg="Uploading artifact \"/workspace/test-logs/unit.json\" as \"228ddfac-29a3-4570-a797-a96ff6c9679c/artifacts/unit.json\""
time="2022-08-31T18:05:23Z" level=info msg="Uploading artifact \"/workspace/test-logs/ci.json\" as \"228ddfac-29a3-4570-a797-a96ff6c9679c/artifacts/ci.json\""
time="2022-08-31T18:05:23Z" level=info msg="Uploading artifact \"/workspace/test-logs/api.json\" as \"228ddfac-29a3-4570-a797-a96ff6c9679c/artifacts/api.json\""
time="2022-08-31T18:05:22Z" level=info msg="Scanning for artifacts..."
make: *** [test-go] Error 1
Makefile:518: recipe for target 'test-go' failed
===================================================
    --- FAIL: TestTeleportClient_Login_local/OTP_device_login_with_hijack (5.00s)
        	Test:       	TestTeleportClient_Login_local/OTP_device_login_with_hijack
        	            	Post "https://localhost:45505/v1/webapi/mfa/login/finish": context deadline exceeded
        	Error:      	Received unexpected error:
        	Error Trace:	api_login_test.go:227
    api_login_test.go:227: 
WARNING: You are using insecure connection to SSH proxy https://localhost:45505
=== RUN   TestTeleportClient_Login_local/OTP_device_login_with_hijack
===================================================
OUTPUT github.com/gravitational/teleport/lib/client.TestTeleportClient_Login_local/OTP_device_login_with_hijack
===================================================
--- FAIL: TestTeleportClient_Login_local (23.64s)
=== RUN   TestTeleportClient_Login_local
===================================================
OUTPUT github.com/gravitational/teleport/lib/client.TestTeleportClient_Login_local
===================================================
FAIL: github.com/gravitational/teleport/lib/client.TestTeleportClient_Login_local
FAIL: github.com/gravitational/teleport/lib/client.TestTeleportClient_Login_local/OTP_device_login_with_hijack
FAIL: github.com/gravitational/teleport/lib/client
===================================================
4068 tests passed. 3 failed, 79 skipped
===================================================
```

**Comment by codingllama** [comment]
Thanks for reporting, Noah.

I went back a good number of commits, all the way to https://github.com/gravitational/teleport/commit/9ca045bacde05bee1f240bf02794309a77da5c41, which was the last time I did work here, but I'm having trouble getting even that revision to work. I'm pretty sure that commit was good, no theories yet as why it's now causing problems.

I'm not sure I'll be able to make progress in this one before the break, so copying @tobiaszheller as well. Maybe you'll have some time and/or manage some progress while I'm out. ;)
